delpay: introduced a new rpc method to delete a payment by hash (#3899)
ElementsProject/lightning: doc/lightning-delpay.7.md

lightning-delpay -- Command for removing a completed payment
============================================================

SYNOPSIS
--------

> **delpay** *payment_hash* \[status\] (is not specify the status is complete)

Reviewer: You have a `partid` as well that is not documented.

Defaults should be described in "DESCRIPTION" section, not in the synopsis.

Author: Updaed

DESCRIPTION
-----------

The **delpay** RPC command removes a payment as given in **listsendpays** or **listpays** with a complete or failed
status. However, the command doesn't permit to remove the pending payment.

> 

Reviewer: DESCRIPTION section needs to describe every parameter, and describe defaults for that parameter as well if it is optional and not specified.

Author: Updated but the doc needs a second review in the next hours (Caused from my Dyslexia problem :smile: )

lightning-delpay -- Command for removing a completed or failed payment
============================================================

SYNOPSIS
--------

> **delpay** *payment_hash* \[status\]

Reviewer: Escape the underscore (as well all instances)

```
*payment\_hash*
```

Author: Changed https://github.com/ElementsProject/lightning/commit/627a374820dd03876eae9e4a99d661cb39f63934

DESCRIPTION
-----------

The **delpay** RPC command removes a payment as given in **listsendpays** or **listpays** with a complete or failed
status. However, the command doesn't permit to remove a pending payment.

- *payment_hash*: Rapresents the unique identifier of a payment. To find it, the caller can run **listpays** or **listsendpays**;

> - *status*: Represents a payment status. Otherwise, if it is not specified it is equal to *complete*.

Reviewer: You should list here the possible values.

```
*status* is the expected status of the payment.
It can be *complete* or *pending*.
Only delete if the payment status matches.
If not specified, defaults to *complete*.
```

Author: Thanks for this suggestion. modify made into https://github.com/ElementsProject/lightning/commit/627a374820dd03876eae9e4a99d661cb39f63934

EXAMPLE JSON REQUEST
------------
```json
{
  "id": 82,
  "method": "delpay",
  "params": {
    "payment_hash": "4fa2f1b001067ec06d7f95b8695b8acd9ef04c1b4d1110e3b94e1fa0687bb1e0",
    "status": "complete"
  }
}
```

RETURN VALUE
------------

On success, the command will return a payment object, such as the **listsendpays**. In addition, if the payment is a MPP (Multi part payment) the command return a list of 
payments; a payment object for each partid.

On failure, an error is returned and any payment is deleted. If the lightning process fails before responding, the
caller should use lightning-listsentpays(7) or lightning-listpays(7) to query whether this payment was deleted or not.

The following error codes may occur:

- -32602: Some parameter missed or some parameter is malformed;

> - 211: Payment with payment_hash have a wrong status. To check the correct status run the command **paystatus**;

Reviewer: ```
*payment\_hash*
```

Author: Changed https://github.com/ElementsProject/lightning/commit/627a374820dd03876eae9e4a99d661cb39f63934

lightning-delpay -- Command for removing a completed or failed payment
============================================================

SYNOPSIS
--------

**delpay** *payment\_hash* \[status\]

DESCRIPTION
-----------

> The **delpay** RPC command removes a payment as given in **listsendpays** or **listpays** with a complete or failed
> status. Deleting a `pending` payment is an error.

Reviewer 2: ```suggestion
The **delpay** RPC command deletes a payment with the given `payment_hash` if its status is either `complete` or `failed`. Deleting a `pending` payment is an error.
```

> status. However, the command doesn't permit to remove a pending payment.

Reviewer 3: ```suggestion
status. Deleting a `pending` payment is an error.
```

> - *payment\_hash*: Rapresents the unique identifier of a payment. To find it, you can run **listpays** or **listsendpays**;

Reviewer 3: ```suggestion
- *payment\_hash*: The unique identifier of a payment. Find in **listpays** or **listsendpays**;
```

> - *status* is the expected status of the payment. It can be *complete* or *failed*. 

Reviewer 3: ```suggestion
- *status*: Expected status of the payment. Valid values are *complete* or *failed*. 
```

> Only deletes if the payment status matches. If not specified, defaults to *complete*.

Reviewer 2: Shouldn't an unspecified value for `status` mean we don't care, and either `complete` or `failed` will be removed?

Reviewer: Since `delpay` is supposed to mirror `delinvoice`, which has the `*status*` as a required arg, maybe it should be a required arg here as well?

Author: Yep I agree with this point! thanks for clarifying!

> Only delete if the payment status matches. If not specified, defaults to *complete*.

Reviewer 3: ```suggestion
Only deletes if the payment status matches. If not specified, defaults to *complete*.
```

EXAMPLE JSON REQUEST
------------
```json
{
  "id": 82,
  "method": "delpay",
  "params": {
    "payment_hash": "4fa2f1b001067ec06d7f95b8695b8acd9ef04c1b4d1110e3b94e1fa0687bb1e0",
    "status": "complete"
  }
}
```

RETURN VALUE
------------

> On success, the command will return a payment object, such as the **listsendpays**. If the payment is aa MPP (multi part payment) the command return a list of 

Reviewer 2: ```suggestion
If successful the command returns a payment object, in the same format as **listsendpays**.
If the payment is a multi-part payment (MPP) the command return a list of 
```

> On success, the command will return a payment object, such as the **listsendpays**. In addition, if the payment is a MPP (Multi part payment) the command return a list of 

Reviewer 3: ```suggestion
On success, the command will return a payment object, such as the **listsendpays**. If the payment is an MPP (multi-part payment) a list of 
```

> payments; a payment object for each partid.

Reviewer 3: ```suggestion
payments will be returned -- one payment object for each partid.
```

> On failure, an error is returned and any payment is deleted. If the lightning process fails before responding, the

Reviewer 3: ```suggestion
On failure, an error is returned. If the lightning process fails before responding, the
```

we don't delete the payment on failure?

caller should use lightning-listsentpays(7) or lightning-listpays(7) to query whether this payment was deleted or not.

The following error codes may occur:

> - -32602: Some parameter missed or some parameter is malformed;

Reviewer 3: ```suggestion
- -32602: Parameter missing or malformed;
```

> - 211: Payment with payment\_hash have a wrong status. To check the correct status run the command **paystatus**;

Reviewer 3: ```suggestion
- 211: Payment status mismatch. Check the correct status via **paystatus**;
```
